__tests__/components/HarvestStatisticsPanel.spec.ts

```diff
@@ -0,0 +1,299 @@
+        expect(getByTestId('total-fees-earned').textContent.trim()).toBe('10.0');
+    });
+
+    test(`harvesting status is auto refreshed to ACTIVE when harvesting status is INPROGRESS_ACTIVATION`, async () => {
+        // Arrange:
+        const dispatch = jest.fn();
+        const { getByTestId, store } = renderComponent(
+            {
+                status: HarvestingStatus.INPROGRESS_ACTIVATION,
+            },
+            dispatch,
+        );
+
+        dispatch.mockImplementation((type: string) => {
+            if (type === 'harvesting/FETCH_STATUS') {
+                store.commit('harvesting/status', HarvestingStatus.ACTIVE);
+            }
+        });
+
+        // Act:
+        await Vue.nextTick();
+        jest.advanceTimersByTime(HARVESTING_STATUS_POLLING_INTERVAL_SECS * 1000);
+        await Vue.nextTick();
+        '';
+        // Assert:
+        expect(getByTestId('status-text').textContent).toBe(i18n.t(`harvesting_status_active`).toString());
+    });
+
+    test(`harvesting status is FAILED`, async () => {
+        // Arrange:
+        const dispatch = jest.fn();
+        const { getByTestId } = renderComponent(
+            {
+                status: HarvestingStatus.FAILED,
+            },
+            dispatch,
+        );
+        jest.spyOn(HarvestingService.prototype, 'getAllHarvestingModels').mockImplementation(() => [currentSignerHarvestingModel]);
+        jest.spyOn(HarvestingService.prototype, 'updateDelegatedHarvestingRequestFailed').mockImplementation(() => {
+            return;
+        });
+
+        // Act:
+        await Vue.nextTick();
+
+        // Assert:
+        expect(getByTestId('status-text').textContent).toBe(i18n.t(`harvesting_status_failed`).toString());
+    });
+
+    test(`harvesting status is set to FAILED when polling limit is reached`, async () => {
+        // Arrange:
+        const dispatch = jest.fn();
+        const { getByTestId, store } = renderComponent(
+            {
+                status: HarvestingStatus.INPROGRESS_ACTIVATION,
+                pollingTrials: HARVESTING_STATUS_POLLING_TRIAL_LIMIT + 1,
+            },
+            dispatch,
+        );
+        jest.spyOn(HarvestingService.prototype, 'getAllHarvestingModels').mockImplementation(() => [currentSignerHarvestingModel]);
+        jest.spyOn(HarvestingService.prototype, 'updateDelegatedHarvestingRequestFailed').mockImplementation(() => {
+            return;
+        });
+
+        // Act:
+        await Vue.nextTick();
+        jest.advanceTimersByTime(HARVESTING_STATUS_POLLING_INTERVAL_SECS * 1000);
+        store.commit('harvesting/status', HarvestingStatus.FAILED);
+        await Vue.nextTick();
+
+        // Assert:
+        expect(getByTestId('status-text').textContent).toBe(i18n.t(`harvesting_status_failed`).toString());
+    });
+
+    test(`refresh status blocks periodically when harvesting status is active`, async () => {
```

Comment:
since you're checking refresh, you might want to check that initial value is set correctly